sdk/appconfiguration/azure-spring-cloud-appconfiguration-config/src/main/java/com/microsoft/azure/spring/cloud/config/AppConfigurationPropertySourceLocator.java

```diff
@@ -0,0 +1,296 @@
+    }
+
+    /**
+     * Creates a new set of AppConfigurationProertySources, 1 per Label.
+     *
+     * @param context Context of the application, part of uniquely define a PropertySource
+     * @param store Config Store the PropertySource is being generated from
+     * @param storeContextsMap the Map storing the storeName -> List of contexts map
+     * @param initFeatures determines if Feature Management is set in the PropertySource.
+     * When generating more than one it needs to be in the last one.
+     * @return a list of AppConfigurationPropertySources
+     */
+    private List<AppConfigurationPropertySource> create(String context, ConfigStore store,
+            Map<String, List<String>> storeContextsMap, boolean initFeatures, FeatureSet featureSet) throws Exception {
+        List<AppConfigurationPropertySource> sourceList = new ArrayList<>();
+
+        try {
+            putStoreContext(store.getEndpoint(), context, storeContextsMap);
+            for (String label : store.getLabels()) {
+                AppConfigurationPropertySource propertySource = new AppConfigurationPropertySource(context, store,
+                        label, properties, clients, appProperties, keyVaultCredentialProvider, keyVaultClientProvider);
+
+                propertySource.initProperties(featureSet);
+                if (initFeatures) {
+                    propertySource.initFeatures(featureSet);
+                }
+                sourceList.add(propertySource);
+            }
+
+            // Setting new ETag values for Watch
+            String watchedKeyNames = clients.watchedKeyNames(store, context);
+            SettingSelector settingSelector = new SettingSelector().setKeyFilter(watchedKeyNames).setLabelFilter("*");
+
+            ConfigurationSetting configurationRevision = clients.getRevison(settingSelector,
+                    store.getEndpoint());
+
+            settingSelector = new SettingSelector().setKeyFilter(FEATURE_STORE_WATCH_KEY).setLabelFilter("*");
+
+            ConfigurationSetting featureRevision = clients.getRevison(settingSelector,
+                    store.getEndpoint());
+
+            String prefix = "_" + context;
+
+            if (configurationRevision != null) {
+                StateHolder.setEtagState(store.getEndpoint() + CONFIGURATION_SUFFIX + prefix, configurationRevision);
+            } else {
+                StateHolder.setEtagState(store.getEndpoint() + CONFIGURATION_SUFFIX + prefix,
+                        new ConfigurationSetting());
+            }
+
+            if (featureRevision != null) {
+                StateHolder.setEtagState(store.getEndpoint() + FEATURE_SUFFIX, featureRevision);
+            } else {
+                StateHolder.setEtagState(store.getEndpoint() + FEATURE_SUFFIX, new ConfigurationSetting());
+            }
+            StateHolder.setLoadState(store.getEndpoint(), true);
+        } catch (RuntimeException e) {
```

Comment:
@mrm9084 In order to avoid occurrence of `REC_CATCH_EXCEPTION` during construction, I am here to modify `Exception` to `RuntimeException`. But I am not clear about the specific role of the code here, should we catch the exception more precise?